### PR TITLE
Adding classic layer tree

### PR DESCRIPTION
**Pull request by weskamm**
This adds a classic layer tree with opacity slider and legend next to the layers in the tree

image

@terrestris/devs please review

### CODE REVIEW

**Comment by hblitza** [comment]
No remarks from my side. Looks good, thanks!

**Comment by weskamm** [comment]
Thanks for review!

File: src/util/AppContextUtil.tsx [hunk 1]
```diff
@@ -73,22 +74,54 @@ class AppContextUtil {
     state.mapView.present.zoom = mapConfig.zoom;
 
     // mapLayers
-    state.mapLayers = union(state.mapLayers, mapLayers);
-    state.mapLayers = AppContextUtil.parseLayers(mapLayers);
+    state.mapLayers = AppContextUtil.parseLayertree(layerTree);
```
Comment (annarieger) [inline src/util/AppContextUtil.tsx:77]: will this break the existing approach with flat layer hierarchie? (lkeyword: ayerset / layer topic)

Comment (weskamm) [reply src/util/AppContextUtil.tsx:77]: to be honest i did not check. i checked the default `static` application and fixed the new shogun-webapp one.
We have an unclear situation here as one can configure layertree and topics at the same time. We need to distinguish that better and need a bigger discussion about all this.

**Pull request merged**